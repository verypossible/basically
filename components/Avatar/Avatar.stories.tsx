import * as React from 'react'

import { boolean, text, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'
import info from '../../utils/storybook-info'
import { H2 } from '../Heading'
import Avatar from './index'

const withInfo = info(`
### Notes

The notes need to be updated for the new component.

### Customization

\`\`\`
import { Avatar } from 'basically'
import styled from 'styled-components'

const MyAvatar = styled(Avatar)\`
  width: 50%;
\`;

<MyAvatar />
\`\`\`
`)

const stories = storiesOf('Components/Avatar', module)
stories.addDecorator(withKnobs)

stories.add('Documentation', withInfo(() => {
  return <Avatar
    round={boolean('round', false)}
    src={text('src', 'https://placeimg.com/640/480/animals')}
    />
}))

stories.add('Different Sizes', () => {
  const BigAvatar = styled(Avatar)`
    width: 15rem;
    height: 15rem
    display: inline-block;
    margin-right: 1rem;
  `
  const MedAvatar = styled(Avatar)`
    width: 10rem;
    height: 10rem
    display: inline-block;
    margin-right: 1rem;
  `
  const RegAvatar = styled(Avatar)`
    display: inline-block;
    margin-right: 1rem;
  `

  return (
    <div>
      <H2>15rem</H2>
      <BigAvatar round />
      <BigAvatar />
      <hr />
      <H2>10rem</H2>
      <MedAvatar round />
      <MedAvatar />
      <hr />
      <H2>Standard Size</H2>
      <RegAvatar round />
      <RegAvatar />
    </div>
  )
})
