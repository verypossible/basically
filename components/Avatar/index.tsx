import * as React from 'react'
import styled from 'styled-components'

interface IAvatar {
  className?: string
  round?: boolean
  src?: string
}

const placeholder = 'https://placeimg.com/640/480/animals'

const UnStyledAvatar = ({ src, ...props }: IAvatar) => {
  return <div {...props} />
}

const roundStyles = `
  border-radius: 50%;
`

const StyledAvatar = styled(UnStyledAvatar)`
  width: 5rem;
  height: 5rem;
  background-image: url('${props => props.src && props.src || placeholder}');
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;

  ${props => props.round && roundStyles}
`

export const Avatar = (props: IAvatar) => <StyledAvatar {...props} />

export default Avatar
