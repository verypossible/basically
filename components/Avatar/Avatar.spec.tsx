import * as React from 'react'

import { shallow } from 'enzyme'
import 'jest-styled-components'

import Avatar from './index'

describe('Avatar', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Avatar />)
    expect(tree).toMatchSnapshot()
  })
})
