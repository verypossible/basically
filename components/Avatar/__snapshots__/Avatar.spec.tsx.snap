// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches the snapshot 1`] = `<Styled(UnStyledAvatar) />`;
